fuzz/fuzz_targets/same_result.rs

```diff
@@ -15,16 +15,44 @@
 
 #![no_main]
 
-use libfuzzer_sys::{arbitrary, fuzz_target};
+use libfuzzer_sys::{
+    arbitrary::{Arbitrary, Unstructured},
+    fuzz_target,
+};
+use wasm_smith::MemoryOffsetChoices;
 use wasmtime::*;
 
 const FUEL: u32 = 1_000;
 
-fuzz_target!(|module: wasm_smith::ConfiguredModule<WasmConfig>| {
+fuzz_target!(|data: &[u8]| {
     let _ = env_logger::try_init();
 
-    let mut module = module;
-    module.ensure_termination(FUEL);
+    let mut u = Unstructured::new(data);
+
+    let mut config = wasm_smith::Config::arbitrary(&mut u).unwrap();
+    config.max_memories = 10;
+
+    // We want small memories that are quick to compare, but we also want to
+    // allow memories to grow so we can shake out any memory-growth-related
+    // bugs, so we choose `2` instead of `1`.
+    config.max_memory32_pages = 2;
+    config.max_memory64_pages = 2;
+
+    // Always generate at least one function that we can hopefully use as an
+    // initialization function.
+    config.min_funcs = 1;
+
+    // Always at least one export, hopefully a function we can use as an
+    // initialization routine.
+    config.min_exports = 1;
+
+    // Always use an offset immediate that is within the memory's minimum
+    // size. This should make trapping on loads/stores a little less
+    // frequent.
+    config.memory_offset_choices = MemoryOffsetChoices(1, 0, 0);
+
+    let mut module = wasm_smith::Module::new(config, &mut u).unwrap();
```

Comment:
Did you try running the fuzz target by chance? I am pretty sure the `unwrap` on `Module::new` at least can fail at runtime. Not sure whether the `Config` one will in practice or not. The `ensure_termination` won't because we aren't enabling the feature in `wasm-smith` that causes it to return an error.

You can run the fuzz target locally to double check via

```shell-session
$ cargo fuzz run --sanitizer none same_result
```

In any case, I think we want to match on error results and simply early return, rather than panicking on error via `unwrap`.

Reply:
so far I only ran the same command CI runs which is `cargo fuzz build --dev -s none`

is it intended that CI only builds this?

Reply:
Yeah it is a trade off because it isn't clear how long is "long enough" to run the fuzzers in CI and just building and linking them already takes a while in CI.